tools/inference_engine/src/impl/model.cpp

```diff
@@ -209,13 +217,30 @@ class FusionVisitor : public GpuVisitor {
         std::cout << "visiting activation...\n";
 
         // check matmul + activation and fuse?
+        // rules:
+        // 1) the last activation in the chain of activations can perform the fusion (remove nodes if needed)
         auto& inputs = pn->get_inputs();
         if (inputs.size() == 1) {
-            auto possibleMatMul = dynamic_cast<GpuMatMul*>(inputs[0]);
-            if (possibleMatMul && possibleMatMul->get_outputs().size() == 1) {
-                std::cout << "possible matmul + activation fusion...\n";
-                possibleMatMul->fuse_with(pn);
-                to_delete_.insert(pn);
+            // is this the last activation in the chain?
```

Comment:
In such functions like this one we shouldn't have any logic/any code which is doing more than just single fusion. 
I think we should have a method to iterate over graph and just fuse single operator, one by one.
Current code could confuse reader and make code a bit harder to read and scale/maintain.